pystiche_papers/sanakoyeu_et_al_2018/_modules.py

```diff
@@ -184,3 +190,58 @@ def residual_block(channels: int) -> ResidualBlock:
     )
 
     return ResidualBlock(residual)
+
+
+def pairwise(iterable: Iterable) -> Iterator[Tuple[Any, Any]]:
+    a, b = itertools.tee(iterable)
+    next(b, None)
+    return zip(a, b)
+
+
+def discriminator_encoder_modules(
```

Comment:
I think we are still talking past each other. I want a `class Discriminator(pystiche.Module)` and a `class DiscriminatorMultiLayerEncoder(enc.MultiLayerEncoder)` that is based on the former.